discord/ui/modal.py

```diff
@@ -25,6 +25,10 @@ class Modal:
     """
 
     def __init__(self, title: str, custom_id: Optional[str] = None) -> None:
+
+        if not (isinstance(custom_id, str) or custom_id is None):
+            raise TypeError(f"expected custom_id to be str, not {type(custom_id).__name__}")
```

Comment:
Same with this one,
`type(custom_id).__name__}`
should be
`custom_id.__class__.__name__`